lib/iris/__init__.py

```diff
@@ -303,7 +312,118 @@ def _load_collection(uris, constraints=None, callback=None):
+        "use_merge",
+        "cat_before_merge",
+        "repeat_until_done",
+    )
+
+    def __init__(
+        self,
+        support_multiple_references: bool = False,
+        multiref_triggers_concatenate: bool = False,
+        use_concatenate: bool = False,
+        use_merge: bool = True,
+        cat_before_merge: bool = False,
+        repeat_until_done: bool = False,
+    ):
+        """Container for loading controls."""
+        self.support_multiple_references = support_multiple_references
+        self.multiref_triggers_concatenate = multiref_triggers_concatenate
+        self.use_concatenate = use_concatenate
+        self.use_merge = use_merge
+        self.cat_before_merge = cat_before_merge
+        self.repeat_until_done = repeat_until_done
+
+    def __repr__(self):
+        msg = (
+            "LoadPolicy("
+            f"support_multiple_references={self.support_multiple_references}, "
+            f"multiref_triggers_concatenate={self.multiref_triggers_concatenate}, "
+            f"use_concatenate={self.use_concatenate}, "
+            f"use_merge={self.use_merge}, "
+            f"cat_before_merge={self.cat_before_merge}, "
+            f"repeat_until_done={self.repeat_until_done}"
+            ")"
+        )
+        return msg
+
+    def copy(self):
+        return LoadPolicy(**{key: getattr(self, key) for key in self._allkeys})
+
+    @contextlib.contextmanager
+    def context(self, policy=None, **kwargs):
+        """Return context manager for temporary options.
+
+        Modifies the given parameters within a context, for the active thread.
+        """
+        # Save the current statr
+        current_state = self.__dict__.copy()
+
+        # Update the state from given policy object and/or method keywords
+        for name in self._allkeys:
+            value = getattr(self, name)
+            if policy and hasattr(policy, name):
+                value = getattr(policy, name)
+            if name in kwargs:
+                value = kwargs[name]
+            setattr(self, name, value)
+
+        try:
+            # Execute the context
+            yield
+        finally:
+            # Return the state
+            self.__dict__.clear()
+            self.__dict__.update(current_state)
+
+
+LOAD_POLICY = LoadPolicy()
+LOAD_POLICY_LEGACY = LoadPolicy()
+LOAD_POLICY_RECOMMENDED = LoadPolicy(
+    support_multiple_references=True, multiref_triggers_concatenate=True
+)
+LOAD_POLICY_COMPREHENSIVE = LoadPolicy(
+    support_multiple_references=True, use_concatenate=True, repeat_until_done=True
+)
+
+
+def _current_effective_policy():
+    policy = LOAD_POLICY
+    if not policy.use_concatenate and policy.multiref_triggers_concatenate:
+        from iris.fileformats.rules import _MULTIREF_DETECTION
+
+        if _MULTIREF_DETECTION.found_multiple_refs:
+            policy = policy.copy()
+            policy.use_concatenate = True
+    return policy
+
+
+def _apply_loading_policy(cubes, policy=None):
```

Comment:
Yes, I agree this could be a useful general tool in its own right.
In that case it would be useful to expose the "POLICY" settings as simple arguments here
(especially if they get simpler, as described in the "new settings arrangement").
Going forward, I'll try + make that workable.